### PR TITLE
Add nutanix disconnected ci job

### PR DESCRIPTION
**Pull request by sgaoshang**

### CODE REVIEW

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 1]
```diff
@@ -2458,6 +2458,19 @@ tests:
...
     workflow: cucushift-installer-rehearse-nutanix-ipi-boot-categories-project
+- as: nutanix-ipi-disconnected-p2-f7
+  cluster: build01
+  cron: 29 2 3,10,17,25 * *
+  steps:
+    cluster_profile: nutanix-qe-dis
+    env:
+      E2E_RUN_TAGS: '@amd64 and @nutanix-ipi and @disconnected and @network-ovnkubernetes
+        and not @fips'
+      TAG_VERSION: '@4.13'
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:2469]: 4.13 -> 4.14

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 2]
```diff
@@ -2853,30 +2853,69 @@ tests:
...
   cluster: build01
-  cron: 33 12 1,8,16,24 * *
+  cron: 47 9 8 * *
   steps:
     cluster_profile: nutanix-qe
+    env:
+      E2E_RUN_TAGS: '@amd64 and @nutanix-ipi and @network-ovnkubernetes and not @fips
+        and @destructive'
+      TAG_VERSION: '@4.13'
+      TEST_ADDITIONAL: Network_Observability
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2865]: I guess we do not need the `TEST_ADDITIONAL` setting here.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 3]
```diff
@@ -2853,30 +2853,69 @@ tests:
...
+- as: nutanix-ipi-disconnected-p3-f28-destructive
+  cluster: build01
+  cron: 49 22 2 * *
+  steps:
+    cluster_profile: nutanix-qe-dis
+    env:
+      E2E_RUN_TAGS: '@amd64 and @nutanix-ipi and @disconnected and @network-ovnkubernetes
+        and not @fips'
+      TAG_VERSION: '@4.13'
+      TEST_ADDITIONAL: Network_Observability
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2891]: I guess we do not need the TEST_ADDITIONAL setting here.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 4]
```diff
@@ -2853,30 +2853,69 @@ tests:
...
   steps:
-    cluster_profile: nutanix-qe
+    cluster_profile: nutanix-qe-dis
     env:
-      E2E_RUN_TAGS: '@amd64 and @nutanix-ipi and @network-ovnkubernetes and not @fips
-        and @destructive'
+      E2E_RUN_TAGS: '@amd64 and @nutanix-ipi and @disconnected and @network-ovnkubernetes
+        and not @fips'
       TAG_VERSION: '@4.13'
       TEST_ADDITIONAL: Network_Observability
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2915]: I guess we do not need the TEST_ADDITIONAL setting here.

Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2915]: I guess we do not need the TEST_ADDITIONAL setting here.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 5]
```diff
@@ -2853,30 +2853,69 @@ tests:
...
   cluster: build01
-  cron: 34 6 26 * *
+  cron: 28 20 16 * *
   steps:
-    cluster_profile: nutanix-qe
+    cluster_profile: nutanix-qe-dis
     env:
-      E2E_RUN_TAGS: '@amd64 and @nutanix-ipi and @network-ovnkubernetes and not @fips
-        and @destructive'
+      E2E_RUN_TAGS: '@amd64 and @nutanix-ipi and @disconnected and @network-ovnkubernetes
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2912]: I think we should use `@proxy` to replace `@disconnected`

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml [hunk 6]
```diff
@@ -512,9 +512,9 @@ tests:
     - ref: cucushift-installer-check
     - chain: openshift-upgrade-qe-test
     workflow: cucushift-installer-rehearse-ibmcloud-ipi-private-fips
-- as: nutanix-ipi-p2-f7
+- as: nutanix-ipi-boot-categories-project-p2-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml:515]: I guess we also need the ci jobs in openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-stable-4.13.yaml

Comment (sgaoshang) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml:515]: Added

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml (hunk 3 above)
Comment (sgaoshang) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml]: So only leave it in nutanix-ipi-boot-categories-project-p3-f28-destructive job?

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml]: Right, FYi, per the commit - 2fa7e4ab694c, seem like some team added the setting in rotate way. If they need, then can file other PR to add it later.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml [hunk 7]
```diff
@@ -523,7 +523,33 @@ tests:
...
+    - ref: cucushift-installer-check
+    - chain: openshift-upgrade-qe-test
+    workflow: cucushift-installer-rehearse-nutanix-ipi-disconnected
+- as: nutanix-ipi-proxy-p2-f28
+  cluster: build01
+  cron: 26 23 4 * *
+  steps:
+    cluster_profile: nutanix-qe-dis
+    env:
+      TEST_FILTERS: ~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml:546]: For cluster behind proxy, I guess we do not need the TEST_FILTERS setting.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 8]
```diff
@@ -2853,30 +2853,68 @@ tests:
...
+  cron: 49 22 2 * *
   steps:
-    cluster_profile: nutanix-qe
+    cluster_profile: nutanix-qe-dis
     env:
-      E2E_RUN_TAGS: '@amd64 and @nutanix-ipi and @network-ovnkubernetes and not @fips
-        and @destructive'
+      E2E_RUN_TAGS: '@amd64 and @nutanix-ipi and @disconnected and @network-ovnkubernetes
+        and not @fips'
+      TAG_VERSION: '@4.13'
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2890]: I guess also need TEST_FILTERS setting here:
```
TEST_FILTERS: ~ChkUpgrade&;~NonPreRelease&;~Serial&;~Disruptive&;~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&
```

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-stable-4.13.yaml [hunk 9]
```diff
@@ -521,7 +521,33 @@ tests:
...
+    - ref: cucushift-installer-check
+    - chain: openshift-upgrade-qe-test
+    workflow: cucushift-installer-rehearse-nutanix-ipi-disconnected
+- as: nutanix-ipi-proxy-p2-f28
+  cluster: build01
+  cron: 2 0 19 * *
+  steps:
+    cluster_profile: nutanix-qe-dis
+    env:
+      TEST_FILTERS: ~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-stable-4.13.yaml:544]: For cluster behind proxy, I guess we do not need the TEST_FILTERS setting.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 10]
```diff
@@ -2458,30 +2458,68 @@ tests:
...
+  cron: 24 2 13 * *
   steps:
-    cluster_profile: nutanix-qe
+    cluster_profile: nutanix-qe-dis
     env:
-      E2E_RUN_TAGS: '@amd64 and @nutanix-ipi and @network-ovnkubernetes and not @fips
-        and @destructive'
+      E2E_RUN_TAGS: '@amd64 and @nutanix-ipi and @disconnected and @network-ovnkubernetes
+        and not @fips'
+      TAG_VERSION: '@4.14'
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:2495]: I guess also need TEST_FILTERS setting here:
```
TEST_FILTERS: ~ChkUpgrade&;~NonPreRelease&;~Serial&;~Disruptive&;~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&
```

**Comment by jianlinliu** [comment]
cc @liangxia to review.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml [hunk 11]
```diff
@@ -523,7 +523,32 @@ tests:
...
+  cron: 22 21 12 * *
+  steps:
+    cluster_profile: nutanix-qe-dis
+    env:
+      TEST_FILTERS: ~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&
+      UPGRADE_CHECK_RUN_TAGS: '@upgrade-check and @4.12 and @nutanix-ipi and not @fips'
+      UPGRADE_PRE_RUN_TAGS: '@upgrade-prepare and @4.12 and @nutanix-ipi and not @fips'
+    test:
+    - ref: cucushift-installer-check
+    - chain: openshift-upgrade-qe-test
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml:538]: Disconnected upgrade use `- chain: openshift-upgrade-qe-test-disconnected`

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 12]
```diff
@@ -2458,30 +2458,69 @@ tests:
...
+    - chain: openshift-e2e-test-qe
+    workflow: cucushift-installer-rehearse-nutanix-ipi-proxy
+- as: nutanix-ipi-proxy-p3-f28-destructive
+  cluster: build01
+  cron: 13 14 4 * *
+  steps:
+    cluster_profile: nutanix-qe-dis
+    env:
+      E2E_RUN_TAGS: '@amd64 and @nutanix-ipi and @proxy and @network-ovnkubernetes
+        and not @fips'
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:2519]: This is a destructive tests. Missing `and @destructive`

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml (hunk 11 above)
Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml:538]: This issue is not fixed

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml [hunk 13]
```diff
@@ -523,7 +523,32 @@ tests:
...
+  cluster: build01
+  cron: 26 23 4 * *
+  steps:
+    cluster_profile: nutanix-qe-dis
+    env:
+      UPGRADE_CHECK_RUN_TAGS: '@upgrade-check and @4.12 and @nutanix-ipi and not @fips'
+      UPGRADE_PRE_RUN_TAGS: '@upgrade-prepare and @4.12 and @nutanix-ipi and not @fips'
+    test:
+    - ref: cucushift-installer-check
+    - chain: openshift-upgrade-qe-test-disconnected
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml:550]: I guess this is placed into a wrong place, for cluster behind proxy, it should be `openshift-upgrade-qe-test`

**Pull request merged**